ksqldb-functional-tests/src/main/java/io/confluent/ksql/test/driver/AssertExecutor.java

```diff
@@ -217,6 +213,33 @@ private static void assertContent(
     }
   }
 
+  private static boolean verifyValues(
+      final KsqlGenericRecord expected,
+      final KsqlGenericRecord actual,
+      final DataSource dataSource,
+      final List<ColumnName> columns
+  ) {
+    final LogicalSchema schema = dataSource.getSchema().withPseudoAndKeyColsInValue(false);
+    for (final ColumnName col : columns) {
+      if (schema.isKeyColumn(col)) {
```

Comment:
For key schema do we also want to check if the no of fields in actual and expected are the same?

Reply:
Added this check for key and value. Also added some tests for asserting tombstones.